ethclient/gethclient/gethclient.go

```diff
@@ -204,6 +220,46 @@ func (ec *Client) SubscribePendingTransactions(ctx context.Context, ch chan<- co
 	return ec.c.EthSubscribe(ctx, ch, "newPendingTransactions")
 }
 
+// TraceCall lets you trace a given eth_call
+func (ec *Client) TraceCall(ctx context.Context, args ethapi.TransactionArgs, blockNrOrHash rpc.BlockNumberOrHash, config *tracers.TraceCallConfig) (interface{}, error) {
+	var result interface{}
+	err := ec.c.CallContext(ctx, &result, "debug_traceCall", args, blockNrOrHash, config)
+	return result, err
+}
+
+// TraceTransaction returns the structured logs created during the execution of EVM
+func (ec *Client) TraceTransaction(ctx context.Context, hash common.Hash, config *tracers.TraceConfig) (interface{}, error) {
+	var result interface{}
+	err := ec.c.CallContext(ctx, &result, "debug_traceTransaction", hash, config)
+	return result, err
+}
+
+// TraceChain subscribes to chain, receiving results from channel BlockTraceResult
+func (ec *Client) TraceChain(ctx context.Context, ch chan<- *BlockTraceResult, start, end rpc.BlockNumber, config *tracers.TraceConfig) (*rpc.ClientSubscription, error) {
+	return ec.c.Subscribe(ctx, "debug", ch, "traceChain", start, end, config)
+}
+
+// TraceBlock returns the structured logs created during the execution of EVM
+func (ec *Client) TraceBlock(ctx context.Context, blob hexutil.Bytes, config *tracers.TraceConfig) ([]*TxTraceResult, error) {
```

Comment:
We shouldn't ask users to use `hexutil.Bytes`. They should work with normal Go primitives preferrably, i.e. `[]byte` here. Then we can cast it ourselves if needed.